requre/online_replacing.py

```diff
@@ -0,0 +1,72 @@
+import sys
+import re
+import functools
+from typing import List, Callable, Optional, Dict
+
+from requre.storage import PersistentObjectStorage
+from requre.import_system import ReplaceType
+
+def update_module(module_name_filter: str,
+                  replacement: Callable, selector: Optional[str] = "",
+                  replace_type: Optional[ReplaceType] = ReplaceType.DECORATOR,
+                  storage_file: Optional[str] = None):
+    def decorator_cover(func):
+        @functools.wraps(func)
+        def _internal(*args, **kwargs):
+            original_storage = None
+            if storage_file:
+                original_storage = PersistentObjectStorage().storage_file
```

Comment:
Packit is more interested in saving and replaying the responses. (The updating of imports is only an implementation detail.)

=> We need to provide nice API exactly for this:

e.g.:

```python
@record(what="hello", in_module="special_requre_module")
def fce():
    pass
```

(There should be a good default for a way of replacement.)

```python
@record_requests()
def fce():
    pass
```

Reply:
yes this is exactly notation what I've choosed: 
`@update_module(module_name_filter=SELECTOR, selector="hello", replacement=decorator_a)`
 what is exactly what you want, reason why you have to add `replacement=` is that requre is generic and there is lots of decorators, so it is up to you what you choose. and your second example one could be just alias for like `record_requests  = record(what="requests", in_module="Session.send", replacement="requre.helpers.request.decorator_plain")

Reply:
Yes, we need to allow setting this, but some shortcuts like `record_requests` can be built on top of it.

Reply:
+1 for a dedicated decorator for python-requests (and also having a generic one)